### ISSUE TITLE
Problem with Cancel button of GenericDialog?

### ISSUE THREAD

**Issue by WC802** [rawpython/remi#504]
Hi! 
I've been trying to use the GenericDialog function but when clicking the Cancel button the following error appears: 'NoneType' object has no attribute 'set_root_widget'. 
Reading the function this seems to come from this lines:
`self._base_app_instance = None`

`self._old_root_widget = None`

I don't know if I'm using it wrong. 
Thank you very much.

**Comment by dddomodossola** [comment]
Hello @WC802 ,

I'm unable to reproduce the problem. Can you show me how you use the GenericDialog?
However an usage example is within the widgets_overview_app.py (https://github.com/rawpython/remi/blob/06808ad6c332f2a4cb3ed33fa834474c66e2d7d9/examples/widgets_overview_app.py#L171)
Note that when you call *dialog.show* you must pass the application instance as argument (https://github.com/rawpython/remi/blob/06808ad6c332f2a4cb3ed33fa834474c66e2d7d9/examples/widgets_overview_app.py#L203).

Have a nice day,
Davide

**Comment by WC802** [comment]
Thank you very much for your answer!
I was able to solve it with the example, which I didn't find earlier. 
Also, I was trying to do a widget for error handling to make a warning pop up with a message that the user can dismiss. This is what GenericDialog does, except it has two buttons. Although I managed to add the class myself, was there already a class that did this? I couldn't find it in the documentation. Or a way to remove the button.
